packages/connection-info/src/connection-secrets.ts

```diff
@@ -190,10 +181,62 @@ export function extractSecrets(connectionInfo: Readonly<ConnectionInfo>): {
   return { connectionInfo: connectionInfoWithoutSecrets, secrets };
 }
 
-function omitPropertiesWhoseValuesAreEmptyObjects<
-  T extends Record<string, Record<string, unknown>>
->(obj: T): Partial<T> {
+function omitPropertiesWhoseValuesAreEmptyObjects<T extends Document>(obj: T) {
   return Object.fromEntries(
     Object.entries(obj).filter(([, value]) => Object.keys(value).length > 0)
   ) as Partial<T>;
 }
+
+const KMS_PROVIDER_SECRET_PATHS = {
+  local: ['key'],
+  aws: ['secretAccessKey', 'sessionToken'],
+  azure: ['clientSecret'],
+  gcp: ['privateKey'],
+};
+
+type AutoEncryptionKMSAndTLSOptions = Partial<
+  Pick<AutoEncryptionOptions, 'kmsProviders' | 'tlsOptions'>
+>;
+
+function extractAutoEncryptionSecrets(data: AutoEncryptionOptions): {
+  data: AutoEncryptionOptions & AutoEncryptionKMSAndTLSOptions;
+  secrets: AutoEncryptionKMSAndTLSOptions;
+} {
+  const secrets: AutoEncryptionKMSAndTLSOptions = {};
+  // Secrets are stored in a kmsProviders and tlsOptions
+  const { kmsProviders, tlsOptions, ...result } = data;
+
+  for (const kmsProviderName in kmsProviders ?? {}) {
+    const key = kmsProviderName.match(/\b(local|aws|gcp|azure)\b(?:)?/)?.[1] as
```

Comment:
Just a question – Does

```suggestion
    const key = kmsProviderName.split(':')[0] as
```

work here?

Currently, this would end up matching something like `kmip:local` as `local`, which hopefully won't be a common case, but maybe this can just be avoided this way?

Reply:
fixed in [3f954e6](https://github.com/mongodb-js/compass/pull/6166/commits/3f954e6b6a3fb85f4943a929b5c551ff8901bb29)